go-controller/pkg/node/linkmanager/link_network_manager.go

```diff
@@ -88,46 +85,36 @@ func (c *Controller) DelAddress(address netlink.Addr) error {
 	if err != nil && !util.GetNetLinkOps().IsLinkNotFoundError(err) {
 		return fmt.Errorf("no valid link associated with addresses %s: %v", address.String(), err)
 	}
+	klog.Infof("Link manager: deleting address %s from link %s", address.String(), link.Attrs().Name)
 	c.mu.Lock()
 	defer c.mu.Unlock()
+	if err := util.GetNetLinkOps().AddrDel(link, &address); err != nil {
```

Comment:
So if delete address fails, im guessing the workqueue will retry. I'm guessing you made this change because in the old code the cache would be updated to remove the address before reconcile ran and deleted things. Would it be better to mark the address as stale or to be deleted in the cache, then let reconcile run and it handle removing the ips? This way you dont rely on any single delete address event, and any reconcile can remove the ip. Then after a successful reconcile, it removes all ips marked as stale in the cache?

Reply:
>So if delete address fails, im guessing the workqueue will retry.

yes and for repair node theres a retry mechanism if there an error. 
Its callers responsibility to handle that now as I didnt want to depend on the cache within link manager because we have no way to rebuilt it anymore upon restart, where as the callers do have a way to rebuild theirs - previously we used labels to track the addresses and could rebuild the cache.